sdk/src/main/scala/com/horizen/account/state/AccountState.scala

```diff
@@ -347,20 +338,31 @@ class AccountState(val params: NetworkParams,
 
   override def getCode(address: Array[Byte]): Array[Byte] = using(getView)(_.getCode(address))
 
-  override def getBaseFee: BigInteger = using(getView)(_.getBaseFee)
-
-  override def getBlockGasLimit: BigInteger = using(getView)(_.getBlockGasLimit)
-
   override def validate(tx: SidechainTypes#SCAT): Try[Unit] = Try {
     tx.semanticValidity()
 
     if (tx.isInstanceOf[EthereumTransaction]) {
 
       val ethTx = tx.asInstanceOf[EthereumTransaction]
-      val blockGasPool = new GasPool(getBlockGasLimit)
+      val blockGasLimit = Account.GAS_LIMIT
+      val blockGasPool = new GasPool(BigInteger.valueOf(blockGasLimit))
+      val blockContext = new BlockContext(
+        // use the null address as forger
+        new Array[Byte](32),
+        // TODO: what timestamp do we use here? just use "now()" or can we get the timestamp of the last block?
+        0,
+        // TODO: what baseFee do we use here? can we get the baseFee of the last block?
```

Comment:
yes, we may use the base fee of the block candidate after current tip.
For this we may store this data in the metadata storage, for example.

Reply:
done